### PR TITLE
Add "card test" mode that tests all cards in a predefined scenario.

### PR DESCRIPTION
**Pull request by bilbo2**
This adds new "cardtest" mode to magarena. (you run it by supplying "--cardtest" arg on commandline)
Essentially, the game takes all implemented tokens and cards and uses them in a simple scenario (short game) to test them.

If some card is flawed, causes crashes or is ineffective, you can see it in the log.

When testing, I found for example a problem with too many lands (5 of each type) and costs having (X) in them - the game thinks too long how to pay for them.

And another problem that was found with this is stack overflow with Kiora's Follower and this scenario (magic.ai.MMAB$MMABWorker.runGame recursively calls itself).

This could be used as kind of quick (can run at most few hours) smoketest after adding a bunch of new cards or making larger changes to some mechanics (if it breaks something, there is some chance you'll encounter the problem in the card test).

Do you think this would be a usable tool during development? Any ideas to make it even better?

### CODE REVIEW

File: src/magic/CommandLineArgs.java [hunk 1]
```diff
@@ -1,220 +1,229 @@
...
+    // sets the desired frames-per-second for the Trident animation library.
+    private Arg<Integer> fps = new Arg<>("--fps", 0, x -> Integer.parseInt(x));
+
+    // limits the number of threads AI uses. (eg. --threads 2)
+    private Arg<Integer> maxAiThreads = new Arg<>("--threads", MagicAI.getMaxThreads(), x -> Integer.parseInt(x));
 
-    private void setDeck1(String deck) {
-        this.deck1 = deck;
+    // Runs specified test game, equivalent to -DtestGame VM arg. (eg. --test TestAura)
+    private Arg<String> testGame = new Arg<>("--threads", "", x -> {
```
Comment (melvinzhang) [inline src/magic/CommandLineArgs.java]: should be "--test" here

**Comment by melvinzhang** [comment]
Yes, this will be a nice way to to an exhaustive test. Currently we rely on the CI to run random games with random sets of cards to test card interactions. Typically it will hit most cards, having a more deterministic way to check cards is a good thing.

File: src/magic/CommandLineArgs.java (hunk 1 above)
Comment (bilbo2) [reply src/magic/CommandLineArgs.java]: fixed

**Comment by bilbo2** [comment]
Types warning were fixed by using Arg<?> .... technically, the code at that place does not know (and does not want to know) what type is stored inside the option.

Added also total time and (average) time per card, printed at end of the test ... perhaps it could be used to track/watch for performance regressions?

**Pull request merged**